app/controllers/games_controller.rb

```diff
@@ -26,7 +39,33 @@ def current_game
     @current_game ||= Game.find(params[:id])
   end
 
-  def game_params
-    params.require(:game).permit(:game_name)
+  def merge_player_color_choice_param
+    if params[:game][:creator_plays_as_black] == '1'
+      { black_user_id: current_user.id }
+    else
+      { white_user_id: current_user.id }
+    end
+  end
+
+  def game_create_params
+    params.require(:game).permit(:game_name, :creator_plays_as_black,
+                                 :white_user_id, :black_user_id)
+      .merge(merge_player_color_choice_param)
```

Comment:
this merge method adds either the white or black user id (as the current user) to the params hash passed to the create action